### PR TITLE
Add a prominent banner to previous and development docs

### LINKED ISSUE #4100: Add prominent banner to old versions of the docs
An example here from Django:

https://docs.djangoproject.com/en/1.7/ref/templates/builtins/

Would suggest something even taller / brighter. Text similar to:
```
This page refers to a past release of Bokeh. For the latest released docs, navigate HERE
```

### PR DESCRIPTION
**Pull request by bryevdv**
issues: fixes #4100

### CODE REVIEW

**Comment by bryevdv** [comment]
I also took this PT to silence some sphinx warnings and errors that had crept in.

**Comment by bryevdv** [comment]
@birdsarah @bokeh/dev any comments welcome, otherwise I will merge late today or early tomorrow.

**Comment by bryevdv** [comment]
Tested `fab latest`

File: sphinx/source/bokeh_theme/static/js/main.js [hunk 1]
```diff
@@ -12,6 +12,21 @@ $(document).ready(function() {
...
+    style = "font-family: sans-serif;font-size: 16px;width: 100%;text-align: center;background-color: pink;color: firebrick;padding: 5px;padding-top:12px; padding-bottom:12px;"
+    loc = window.location.pathname;
+    dev = /^\/en\/dev/
+    if (dev.exec(loc)) {
+      content = $('<div>The documentation on this page refers to a DEVELOPMENT VERSION. For the latest release, go to <a href="http://bokeh.pydata.org/en/latest/">http://bokeh.pydata.org/en/latest/</a></div>').attr('style', style);
+      $('html:first').prepend(content);
+    }
+    else if (v != window._BOKEH_CURRENT_VERSION) {
+      content = $('<div>The documentation on this page refers to a PREVIOUS VERSION. For the latest release, go to <a href="http://bokeh.pydata.org/en/latest/">http://bokeh.pydata.org/en/latest/</a></div>').attr('style', style);
+      $('html:first').prepend(content);
```
Comment (birdsarah) [inline sphinx/source/bokeh_theme/static/js/main.js]: I've actually never used the `:first` pseudo-selector before and had to look it up. According to https://developer.mozilla.org/en-US/docs/Web/CSS/:first it's not supported on firefox. 

What about just  `$('body').prepend(content);`?

**Comment by birdsarah** [comment]
There seem to be unrelated changes in here, like the ellipse stuff.

This is big an improvement, thanks.

**Comment by birdsarah** [comment]
Random thought: should we open a follow-on issue to have it try to take you to the same page on the latest version? e.g. if you're on http://bokeh.pydata.org/en/0.11.0/docs/reference/models/formatters.html it points you to http://bokeh.pydata.org/en/0.11.1/docs/reference/models/formatters.html

File: sphinx/source/bokeh_theme/static/js/main.js (hunk 1 above)
Comment (bryevdv) [inline sphinx/source/bokeh_theme/static/js/main.js]: I did just try it and it worked fine at least on the FF I have installed

**Comment by bryevdv** [comment]
I thought about that, but it also occurred to me that the old page may no longer exist in the new docs so agree that a better 404 is needed first

**Comment by bryevdv** [comment]
> There seem to be unrelated changes in here, like the ellipse stuff.

Yes, there were 5-6 sphinx warnings and errors that crept in, the changes to fix them were tiny, so I just put them in here since I was already working in the docs.

**Comment by bryevdv** [comment]
@birdsarah made the change you recommended and tested (see http://bokeh.pydata.org/en/dev/)  will merge when green unless there are other comments.

**Comment by bryevdv** [comment]
Another possible issue to make: we could put in a check that the docs build has no errors or warnings, so that the little things don't collect.

**Comment by birdsarah** [comment]
> Yes, there were 5-6 sphinx warnings and errors that crept in, the changes to fix them were tiny, so I just put them in here since I was already working in the docs. 

GREAT - I noticed these the other day.

> Another possible issue to make: we could put in a check that the docs build has no errors or warnings, so that the little things don't collect. 

+1

**Pull request merged**